libs/feature/search/src/lib/utils/service/fields.ts

```diff
@@ -220,13 +234,78 @@ if (doc.containsKey('licenseObject.default.keyword') && doc['licenseObject.defau
 if (doc.containsKey('MD_LegalConstraintsUseLimitationObject.default.keyword') && doc['MD_LegalConstraintsUseLimitationObject.default.keyword'].length > 0)
   raw += doc['MD_LegalConstraintsUseLimitationObject.default.keyword'].join('|').toLowerCase();
 
-if (raw.contains('odbl')) emit('ODbL');
-if (raw.contains('pddl')) emit('PDDL');
-if (raw.contains('odc-by')) emit('ODC-By');
-if (raw.contains('cc0') || raw.contains('cc-0')) emit('CC-0');
-if (raw.contains('cc-by') || raw.contains('cc by')) emit('CC BY');
-if (raw.contains('open license')) emit('Open License');
-if (raw.contains('etalab')) emit('Etalab');`
+boolean unknown = true;
+if (raw.contains('pddl') || raw.contains('public domain dedication and licence')) {
+  unknown = false;
+  emit('pddl');
+}
+if (raw.contains('odbl') || raw.contains('open database license')) {
+  unknown = false;
+  emit('odbl');
+}
+if (raw.contains('odc-by') || raw.contains('opendatacommons.org/licenses/by/summary/"')) {
+  unknown = false;
+  emit('odc-by');
+}
+
+if (raw.contains('cc-by-sa') || raw.contains('creative commons attribution share-alike')) {
+  unknown = false;
+  emit('cc-by-sa');
+} else if (raw.contains('cc-by') || raw.contains('cc by') || raw.contains('creative commons attribution')) {
+  unknown = false;
+  emit('cc-by');
+} else if (raw.contains('cc0') || raw.contains('cc-0') || raw.contains('cczero') || raw.contains('cc-zero')) {
+  unknown = false;
+  emit('cc-zero');
+}
+
+if (raw.contains('etalab') && (raw.contains('v2') || raw.contains('2.0'))) {
+  unknown = false;
+  emit('etalab-v2');
+} else if (raw.contains('open licence') || raw.contains('licence ouverte') || raw.contains('licence_ouverte')) {
+  unknown = false;
+  emit('etalab');
+}
+
+if(unknown) emit('unknown');`
     )
   }
+
+  getAvailableValues(): Observable<FieldAvailableValue[]> {
```

Comment:
We should decouple the parent method `getAvailableValues` to fetch/process logic cause here you have to copy paste the fetch logic just to change the process one.
Or implement in the parent a `mapResponse()` and a `mapTranslatedResponse()` or something like this

Reply:
Done in the latest commit